### PR TITLE
Load LDAP settings on each authentication attempt

### PR DESCRIPTION
**Pull request by edmundoa**
I have worked on a fix for #934, but I'm not really familiar with that part of the code, so I think it's best if somebody takes a look at it before merging into 1.0. But don't be afraid, you will be rewarded with a cat picture for your effort!!1!

### CODE REVIEW

**Comment by kroepke** [comment]
lgtm.

Originally the cached version was used because we asked for LDAP settings more often, now we have sessions and its only used on explicit login/session creation.

Feel free to merge.

File: graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java [hunk 1]
```diff
@@ -20,40 +20,35 @@
 import org.apache.directory.api.ldap.model.exception.LdapException;
 import org.apache.directory.ldap.client.api.LdapConnectionConfig;
 import org.apache.directory.ldap.client.api.LdapNetworkConnection;
-import org.apache.shiro.authc.AuthenticationException;
-import org.apache.shiro.authc.AuthenticationInfo;
-import org.apache.shiro.authc.AuthenticationToken;
-import org.apache.shiro.authc.SimpleAccount;
-import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.authc.*;
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java]: Please configure your IDE to not use wildcard imports.

(in IntelliJ IDEA: Settings -> Java -> Code Style -> Imports -> Class count to use import with '*')

Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java]: Thank you! Will change the import style and merge the result into 1.0.

**Comment by edmundoa** [comment]
Thank you for looking into it! I applied the style changes @joschi recommended and the fix is now in the 1.0 branch (d93cc012c0f14dae55137aee9381eced52765e9b).

As promised:
laughing-cat

**Comment by joschi** [comment]
WAT

**Pull request closed without merging**